go/vt/mysqlctl/backup.go

```diff
@@ -263,7 +263,7 @@ func Restore(
 		// There are no backups (not even broken/incomplete ones).
 		logger.Errorf("no backup to restore on BackupStorage for directory %v. Starting up empty.", dir)
 		// Since this is an empty database make sure we start replication at the beginning
-		if err = mysqld.ResetReplication(ctx); err == nil {
+		if err = mysqld.ResetReplication(ctx); err != nil {
```

Comment:
The previous version is correct. If there is an error from ResetReplication we want to report that, otherwise we want to report ErrNoBackup.

Reply:
The logic here is incorrect. What it is saying is that if ResetReplication succeeeds we report an error and set ErrNoBackup. This should either be `== nil` and the log message needs to change or we sohuld remove the `err = ErrNoBackup` to not be gated by this condition, and we should change the condition to be err2. 

Since I think that is what we want I will make those changes for review

Reply:
This is not right. As it is now, we will never return an ErrNoBackup because err is immediately overwritten by the result of the call to `mysqld.ResetReplication`. I think the only change needed from the earlier version is to add the log statement.
What we want is: 
- if ResetReplication throws an error, report that
- if PopulateMetadataTables throws an error, report that
- otherwise report ErrNoBackup

Reply:
Ok so in general if we have no backup, I will set it to try and Reset Replication, and try to PopulateMetadata, report any errors if it encounters them, then I will just return ErrNoBackup. I am going to refactor a bit to make this more explicit.